metal/classifier.py

```diff
@@ -270,45 +334,88 @@ def _set_scheduler(self, scheduler_config, optimizer):
 
     def score(
         self,
-        X,
-        Y,
+        data,
         metric=["accuracy"],
         break_ties="random",
         verbose=True,
+        print_confusion_matrix=True,
         **kwargs,
     ):
         """Scores the predictive performance of the Classifier on all tasks
 
         Args:
-            X: The input for the predict method
-            Y: An [n] or [n, 1] torch.Tensor or np.ndarray of target labels in
-                {1,...,k}
+            data: a Pytorch DataLoader, Dataset, or tuple with Tensors (X,Y):
+                X: The input for the predict method
+                Y: An [n] or [n, 1] torch.Tensor or np.ndarray of target labels
+                    in {1,...,k}
             metric: A metric (string) with which to score performance or a
                 list of such metrics
             break_ties: A tie-breaking policy (see Classifier._break_ties())
             verbose: The verbosity for just this score method; it will not
-                update the class config
+                update the class config.
+            print_confusion_matrix: Print confusion matrix
 
         Returns:
             scores: A (float) score or a list of such scores if kwarg metric
                 is a list
         """
-        Y = self._to_numpy(Y)
-        Y_p = self.predict(X, break_ties=break_ties, **kwargs)
+        Y_pred, Y = self._get_predictions(data, break_ties=break_ties, **kwargs)
```

Comment:
I understand the desire to have Y_pred instead of Y_p so that people don't confuse the p to mean "probabilistic", for example. But if we're going to make this change, I think we just need to be hard committed: all Y predictions everywhere must be Y_pred instead of Y_p. Looks like you changed it in a bunch of places, so that's probably what you're aiming for. If so, go ahead and update the design doc as well to reflect that. (And I'll add it as a TODO of mine to get the design doc out of a gdoc and into the repo so it's tracked). This change does get rid of the scheme for combining multiple single letter suffixes (e.g., Y_ps is predicted soft labels), but probably not a big deal, and I do like the clarity of Y_pred.

Reply:
I'll revert to `Y_p` for now- can update terminology in a separate PR if we want.  Good catch!

Reply:
Also another reason to have the design doc be tracked in the repo too- we should do that first!